lib/Test/Nginx/Util.pm

```diff
@@ -728,32 +728,25 @@ sub run_tests () {
 
 sub setup_server_root () {
     if (-d $ServRoot) {
+        # Take special care, so we won't accidentally remove
+        # real user data when TEST_NGINX_SERVROOT is mis-used.
+        remove_tree($ConfDir, $HtmlDir, $LogDir, glob("$ServRoot/*_cache"),
+                    glob("$ServRoot/*_temp"));
```

Comment:
This does not look right. You already remove the nginx.pid file which we want to keep in the HUP reload mode.

Reply:
@agentzh 
Fixed in the new commit.